src/toast/unported/todmap/atm.py

```diff
@@ -1053,18 +1053,29 @@ def load_realization(self):
         if self._rank == 0:
             log = Logger.get()
 
-            hf = h5py.File(cachefile, "r")
-            # Read metadata
-            meta = hf.attrs
-            for k, tp in self._meta_keys():
-                # Copy the metadata value into a dictionary, casting to correct type
-                mdata[k] = tp(meta[k])
-            log.debug("Loaded metadata for {}".format(rname))
+            try:
+                hf = h5py.File(cachefile, "r")
+                # Read metadata
+                meta = hf.attrs
+                for k, tp in self._meta_keys():
+                    # Copy the metadata value into a dictionary, casting to correct type
+                    mdata[k] = tp(meta[k])
+                log.debug("Loaded metadata for {}".format(rname))
+            except Exception as e:
+                print(
+                    f"ERROR: failed to load cached atmosphere realization"
+                    f" from {cachefile}. Will simulate again.",
+                    flush=True,
+                )
+                mdata = None
```

Comment:
Obviously this is in a temporary directory of toast2 code which will be deleted shortly.  If it is important to backport this to toast2, then we should open a PR against master.

Reply:
Not important enough to backport. I did make sure the feature is now included in TOAST3.